java/src/jmri/util/ImmediatePipeOutputStream.java

```diff
@@ -0,0 +1,34 @@
+package jmri.util;
+
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+
+/**
+ * Makes a workaround for standard {@link PipedOutputStream} wait.
+ * <p>The {@link PipedInputStream#read()}, in case the receive buffer is
+ * empty at the time of the call, waits for up to 1000ms.
+ * {@link PipedOutputStream#write(int)} does call <code>sink.receive</code>,
+ * but does not <code>notify()</code> the sink object so that read's
+ * wait() terminates.
+ * </p><p>
+ * As a result, the read side of the pipe waits full 1000ms even though data
+ * become available during the wait.
+ * </p><p>
+ * The workaround is to simply {@link PipedOutputStream#flush} after write,
+ * which returns from wait()s immediately.
+ * </p>
+ */
+public class ImmediatePipeOutputStream extends PipedOutputStream {
```

Comment:
Can you rename this class from `ImmediatePipeOutputStream` to `ImmediatePipedOutputStream` (_Pipe**d**_ instead of _Pipe_)?